generator/testdata/google/example/library/v1/library_service.gapic.cc.baseline

```diff
@@ -9,84 +11,192 @@
 gax::StatusOr<::google::example::library::v1::Book>
 LibraryService::CreateBook(
 ::google::example::library::v1::CreateBookRequest const& request) {
-  // TODO: actual useful work, e.g. retry, backoff, metadata, pagination, etc.
+  // TODO: update metadata and idempotency, etc.
   grpc::ClientContext context;
   ::google::example::library::v1::Book response;
   grpc::Status status = stub_->CreateBook(&context, request, &response);
+
   if (status.ok()) {
     return response;
-  } else {
-    return gax::GrpcStatusToGaxStatus(status);
   }
+
+  std::string error_message;
+  // TODO: use idempotency policy to determine whether a failure is permanent
+  if (!isPermanentFailure(status)) {
```

Comment:
I think this can be handled by the retry_policy. If the purpose of this function is to make decisions based on the idempotency, then it doesn't need the status object, it needs info about the method being called. So I would say that the decision should be made inside the retry_policy->OnFailure method, which will have access to a gax::CallContext object.